explanation.md

```diff
@@ -0,0 +1,38 @@
+#Explanation of spy function
+
+##Basic structure
+
+The basic structure of the file is as follows:
+
+```
+var firstAnonFunc = function(global, factory){}
+var secondAnonFunc = function(fn, ctx){ return spyfn }
+firstAnonFunc(this, secondAnonFunc);
+```
+Using the (function(){}()) notation, the first anonymous function is executed as it is defined. This function is passed the global context held in the `this` variable and another anonymous function which in turn will generate the spy function whenever it is called.
+
+##The first anonymous function
+The role of the first anonymous function is to make the second anonymous function callable from any point in our script. This is done in one of three ways depending on the version of JavaScript. The simplest way is to simply assign the second anonymous function to a global variable named `spyOn`.
+
+```
+global.spyOn = factory
+``` 
+Remember, `factory` is just the second anonymous function. Now that function is globally accessible through `spyOn`.
+
+##Second anonymous function
+When `spyOn` is called it should be passed the function that you want to spy on. It will then return this function augmented with a few methods. i.e.
+
+```
+myFuncSpiedOn = spyOn(myFunc)
+```
+
+If myFuncSpiedOn is called it will return `fn.apply(ctx, arguments)`. In other words, it will execute `fn` (the function passed to `spyOn`) with context `ctx` (if it was passed as the second argument of `spyOn`) and the arguments of `fn`.
```

Comment:
it is going to pass the arguments with what you are calling your `myFuncSpiedOn` to `fn`.